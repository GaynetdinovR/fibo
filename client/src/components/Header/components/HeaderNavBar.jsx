import styles from "../../../styles/components/Header.module.sass";
import Button from "../../../ui/Button.jsx";
import LinkInList from "../../../ui/LinkInList.jsx";
import PhoneNumber from "../../../ui/PhoneNumber.jsx";
import { useLocation } from "react-router-dom";
import RouterLinkInList from "../../../ui/RouterLinkInList.jsx";
import { useSelector } from "react-redux";

const HeaderNavBar = ({ isMenuOpen, setMenu, setAuth }) => {
	const user = useSelector((state) => state.user);

	const navbarLinks = [
		{ text: "Пиццы", href: "#pizza", to: "/" },
		{ text: "Пасты", href: "#pasta", to: "/" },
		{ text: "Супы", href: "#soup", to: "/" },
		{ text: "Салаты", href: "#salad", to: "/" },
		{ text: "Закуски", href: "#snack", to: "/" },
		{ text: "Напитки", href: "#drink", to: "/" },
		{ text: "Акции", to: "/promo" },
		{ text: "Контакты", to: "/contacts" },
		{ text: "Профиль", to: "/profile", isNeedLogIn: true }
	];

	const windowWidth = window.innerWidth;

	const cartBtn = (
		<Button className={styles.header__cart_btn}>Корзина | 0</Button>
	);

	const phoneNumber = (
		<PhoneNumber
			phoneNumber={"8 499 391-84-49"}
			className={styles.header__phone_number}
		/>
	);

	const logInBtn = (
		<button
			onClick={() => {
				setAuth(true);
				setMenu(false);
			}}
			className={styles.header__auth}
		>
			Войти
		</button>
	);

	return (
		<nav
			className={
				isMenuOpen || windowWidth > 960
					? styles.header__navbar
					: styles.header__navbar_closed
			}
		>
			<ul className={styles.header__list} onClick={() => setMenu(false)}>
				{navbarLinks.map((item, i) => {
					if (item.isNeedLogIn) {
						if (!user.isLogged) return;
					}

					if (useLocation().pathname === item.to) {
						return (
							<LinkInList
								key={i}
								liClass={styles.header__item}
								aClass={styles.header__link}
								href={item.href}
							>
								{item.text}
							</LinkInList>
						);
					} else {
						return (
							<RouterLinkInList
								key={i}
								liClass={styles.header__item}
								aClass={styles.header__link}
								href={item.href}
								to={item.to}
							>
								{item.text}
							</RouterLinkInList>
						);
					}
				})}
			</ul>

			<div className={styles.header__right_bottom_side}>
				{!user.isLogged ? logInBtn : null}
				{windowWidth > 960 ? cartBtn : null}
			</div>
			{windowWidth <= 480 ? phoneNumber : null}
			{windowWidth <= 480 ? cartBtn : null}
		</nav>
	);
};

export default HeaderNavBar;
