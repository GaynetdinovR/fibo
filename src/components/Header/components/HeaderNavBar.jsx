import styles from "../../../styles/components/Header.module.sass";
import Button from "../../../ui/Button.jsx";

const HeaderNavBar = ({ isMenuOpen }) => {
	const navbarLinks = [
		"Пицца",
		"Паста",
		"Супы",
		"Салаты",
		"Напитки",
		"Десерты",
		"Бакалея",
		"Антипасти",
		"Акции",
		"Комбо",
		"Контакты"
	];
	const windowWidth = window.innerWidth;

	const cartBtn = (
		<Button className={styles.header__cart_btn}>Корзина | 0</Button>
	);

	const phoneNumber = (
		<a href="tel:84993918449" className={styles.header__phone_number}>
			8 499 391-84-49
		</a>
	);

	return (
		<nav
			className={
				isMenuOpen || windowWidth > 768
					? styles.header__navbar
					: styles.header__navbar_closed
			}
		>
			<ul className={styles.header__list}>
				{navbarLinks.map((item, i) => (
					<li key={i} className={styles.header__item}>
						<a className={styles.header__link}>{item}</a>
					</li>
				))}
			</ul>
			<button className={styles.header__auth}>Войти</button>
			{windowWidth <= 480 ? phoneNumber : ""}
			{windowWidth > 960 || windowWidth <= 480 ? cartBtn : ""}
		</nav>
	);
};

export default HeaderNavBar;
