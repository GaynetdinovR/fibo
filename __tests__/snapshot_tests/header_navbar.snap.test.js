import renderer from "react-test-renderer";
import { expect, it } from "@jest/globals";
import HeaderNavBar from "../../src/components/Header/components/HeaderNavBar.jsx";

it("renders correctly", () => {
	const tree = renderer.create(<HeaderNavBar />).toJSON();
	expect(tree).toMatchSnapshot();
});
