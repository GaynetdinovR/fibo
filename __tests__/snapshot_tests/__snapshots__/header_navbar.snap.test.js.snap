// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="header__navbar"
>
  <ul
    className="header__list"
  >
    <li
      className="header__item link_in_list__item"
    >
      <a
        className="header__link link_in_list__link"
      >
        Пицца
      </a>
    </li>
    <li
      className="header__item link_in_list__item"
    >
      <a
        className="header__link link_in_list__link"
      >
        Паста
      </a>
    </li>
    <li
      className="header__item link_in_list__item"
    >
      <a
        className="header__link link_in_list__link"
      >
        Супы
      </a>
    </li>
    <li
      className="header__item link_in_list__item"
    >
      <a
        className="header__link link_in_list__link"
      >
        Салаты
      </a>
    </li>
    <li
      className="header__item link_in_list__item"
    >
      <a
        className="header__link link_in_list__link"
      >
        Напитки
      </a>
    </li>
    <li
      className="header__item link_in_list__item"
    >
      <a
        className="header__link link_in_list__link"
      >
        Десерты
      </a>
    </li>
    <li
      className="header__item link_in_list__item"
    >
      <a
        className="header__link link_in_list__link"
      >
        Бакалея
      </a>
    </li>
    <li
      className="header__item link_in_list__item"
    >
      <a
        className="header__link link_in_list__link"
      >
        Антипасти
      </a>
    </li>
    <li
      className="header__item link_in_list__item"
    >
      <a
        className="header__link link_in_list__link"
      >
        Акции
      </a>
    </li>
    <li
      className="header__item link_in_list__item"
    >
      <a
        className="header__link link_in_list__link"
      >
        Комбо
      </a>
    </li>
    <li
      className="header__item link_in_list__item"
    >
      <a
        className="header__link link_in_list__link"
      >
        Контакты
      </a>
    </li>
  </ul>
  <button
    className="header__auth"
  >
    Войти
  </button>
  <button
    className="header__cart_btn default_button"
  >
    Корзина | 0
  </button>
</nav>
`;
